src/demos_scheduler.hpp

```diff
@@ -108,6 +108,7 @@ class DemosScheduler
         logger->info("Starting scheduler");
         memory_tracker::enable();
         auto start_time = std::chrono::steady_clock::now();
+        Slice::start_time = start_time;
```

Comment:
I don't like this global. It's still not ideal, but I'd prefer moving it inside the SchedulerLogger class so it doesn't pollute Slice.

Reply:
Too much work for now. I need to resolve #68 without breaking current behavior. We can make the code nice later. But given that it's used only in debug builds, it has low priority.